File: src/main/java/io/scriptor/gdb/ClientState.java
package io.scriptor.gdb;

import java.nio.ByteBuffer;

public final class ClientState {

    public final ByteBuffer buffer = ByteBuffer.allocateDirect(0x2000);
    public final StringBuilder packet = new StringBuilder();
    public final StringBuilder packetChecksum = new StringBuilder();

    public boolean append = false;
    public boolean end = false;

    public int checksum;

    public boolean noack = false;
}

File: src/main/java/io/scriptor/gdb/ClientState.kt
package io.scriptor.gdb

import java.nio.ByteBuffer

class ClientState {

    val buffer: ByteBuffer = ByteBuffer.allocateDirect(0x2000)
    val packet: StringBuilder = StringBuilder()
    val packetChecksum: StringBuilder = StringBuilder()

    var append: Boolean = false
    var end: Boolean = false

    var checksum: UInt = 0u

    var noack: Boolean = false

    var stopId: Int = -1
    var stopCode: UInt = 0x00u
}
